### DISCUSSION TITLE
Ejemplo

### DISCUSSION THREAD

**Discussion by mmgmanuel** [Barnetik/tbai-php-lib#9] [Q&A]
Buenos días,

¿Podrías indicar algunos ejemplos de su uso?

Gracias y un saludo.

**Comment by arraintxo** [comment]
Buenas,

a día de hoy lo mejor para entender como funciona la librería es echar un vistazo a los tests, ya sé que no es la mejor de las respuestas pero apenas tenemos tiempo para documentarlo mejor por ahora. 

La clase TicketMother en los tests se encarga de generar las facturas de prueba y desde los tests de Endpoints puedes ver como se envían a los servidores de la administración pública.

**Comment by Mrcat926** [comment]
¿Hola, podría describir cómo usar este programa?

**Comment by lozal2244** [comment]
Hola, 

Con todo el respeto, los tests no son nada claros, hay varios archivos que no entiendo que hacen cada uno, no veo un claro ejemplo e un envio de un ticketbai, solo veo clases que llaman a otras y certificados que se llaman unos a otros.

Un solo archivito en php y un mini menu sería muy util para saber como usar cada cosa, o al menos un pdf con una guia simple de uso, el mayor problema de la integración con ticket bai esta siendo este embudo que hay con la documentación, tengo cientos de clientes preocupados porque no hay forma de integrarnos con vosotros.

Ademas las tres adminsitraciones no se ponen de acuerdo para un único formato de ticket, cuando consigues hablar con alguien al telefono no sabe información del resto de administraciones y te remiten a documentaciones que no entienden por lo que quedamos desatentidos totalmente.

LEs escribo desde www.pos-tpv.com CEO de la empresa, llevamos meses con este tema y se acerca ya la fecha de alta con ticket BAI y no hay manera de conseguir algo de ayuda.

Un saludo,
Alex Lozano

**Comment by arraintxo** [comment]
Buenos días Alex ( @lozal2244 ),

siento mucho que los tests no os sean suficientemente claros, la verdad es que la intención de los mismos no es documentar la librería, sino testear que realmente funciona, pero siempre es posible tirar un poco de su código para aprender a utilizarla. La clase TicketBaiMother tiene un par de ejemplos de como generar una factura. Y los endpoints bajo el apartado de API tiene ejemplos de como presentarlas para cada una de las diputaciones.

El tema de la documentación de la librería es un problema de la cantidad de recursos limitados que podemos dedicar a la misma. Es una librería que hemos desarrollado sin ningún tipo de ayuda económica y la hemos liberado bajo licencia GPL para que los proyectos compatibles con esta licencia puedan utilizarla (no estoy seguro de que sea vuestro caso, si no es así os animo a que os pongáis en contacto con nosotros).

Respecto a las quejas sobre el funcionamiento de las administraciones,  esta no es una implementación oficial provista por las diputaciones (no existe tal cosa), por lo que no nos corresponde recibirlas. Somos una cooperativa completamente independiente que nada tiene que ver con la administración pública.

Un saludo,

**Comment by lozal2244** [comment]
Buenas tardes,
Disculpen porque no sabía que no perteneciía a la adminsitración. En enlace a esta libreria me lo proporcionó precisamente alguien de la adminsitración pública por lo que entendí que era parte de sus desarrollos.

Intentaré en ese caso apañarme con lo que hay y solo me queda agradecerles el trabajo y que lo liberen en licencia GPL, mil gracias :)
